ipn/ipnlocal/network-lock.go

```diff
@@ -125,15 +127,10 @@ func (b *LocalBackend) tkaSyncIfNeeded(nm *netmap.NetworkMap) error {
 			}
 			isEnabled = true
 		} else if !wantEnabled && isEnabled {
-			if b.tka.authority.ValidDisablement(bs.DisablementSecret) {
-				b.tka = nil
-				isEnabled = false
-
-				if err := os.RemoveAll(b.chonkPath()); err != nil {
-					return fmt.Errorf("os.RemoveAll: %v", err)
-				}
+			if err := b.tkaApplyDisablementLocked(bs.DisablementSecret); err != nil {
+				b.logf("Disablement failed, leaving TKA enabled. Error: %v", err)
```

Comment:
Should this be a returned error rather than just logged?

Reply:
I've opted to log here because returning early would mean that sync is not completed. I know holdbacks are technically out of scope, but the less the better. I've added a comment.